src/test/scala/cromwell/util/SampleWdl.scala

```diff
@@ -1668,4 +1668,32 @@ object SampleWdl {
       """.stripMargin
     override val rawInputs = Map.empty[String, String]
   }
+
+  /**
+    * Outputs which reference themselves.
+    */
+  object InputsAndOutputsAutoReferentiation extends SampleWdl {
```

Comment:
I'm happy to say declaring something twice should be a syntax error or picked up at validation time.